framework/core/src/Admin/WhenSavingSettings.php

```diff
@@ -36,14 +36,18 @@ public function beforeSave(Saving $event): void
     {
         if (array_key_exists('safe_mode_extensions', $event->settings)) {
             $safeModeExtensions = json_decode($event->settings['safe_mode_extensions'] ?? '[]', true);
+            $sorted = [];
 
-            $extensions = $this->extensions->getExtensions()->filter(function ($extension) use ($safeModeExtensions) {
-                return in_array($extension->getId(), $safeModeExtensions);
-            });
+            if ($safeModeExtensions) {
+                $extensions = $this->extensions->getExtensions()->filter(function ($extension) use ($safeModeExtensions) {
+                    return in_array($extension->getId(), $safeModeExtensions);
+                });
 
-            $sorted = array_map(fn (Extension $e) => $e->getId(), $this->extensions->sortDependencies($extensions->all()));
+                $sorted = array_map(fn (Extension $e) => $e->getId(), $this->extensions->sortDependencies($extensions->all()));
+                $sorted = array_values($sorted);
+            }
 
-            $event->settings['safe_mode_extensions'] = json_encode(array_values($sorted));
+            $event->settings['safe_mode_extensions'] = json_encode($sorted);
         }
```

Comment:
does this part throw an error like the config one? I would assume it wouldn't since we do `$event->settings['safe_mode_extensions'] ?? '[]'` here 🤔

Reply:
`safe_mode_extensions` always includes in Saving event with empty value (`safe_mode_extensions
: 
""` in the Advanced page, so it doesn't cause errors here

Reply:
It may looks like this

```php
// $event->settings['safe_mode_extensions'] is '' if no extensions selected
$safeModeExtensions = ! empty($event->settings['safe_mode_extensions']) ? $event->settings['safe_mode_extensions'] : '[]';
$safeModeExtensions = json_decode($event->settings['safe_mode_extensions'], true);
```

Reply:
Ah that makes sense!